### PR TITLE
feat: add currency beacon realtime provider

### PR DESCRIPTION
**Pull request by dolcalmi**
Add https://currencybeacon.com/ realtime provider in order to provide ~multiple~ display currencies. It provides 5,000 Requests/month in free plan

if we want to use wildcard config we need to merge https://github.com/GaloyMoney/price/pull/35 first

### CODE REVIEW

**Comment by nicolasburtey** [comment]
why do you want to provide multiple display currencies?

**Comment by dolcalmi** [comment]
> why do you want to provide multiple display currencies?

translation issue.. this is just a FIAT price provider :)

File: realtime/src/services/exchanges/currencybeacon/index.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  IExchangeService | ExchangeServiceError
+> => {
+  if (!config || !config.apiKey) return new InvalidExchangeConfigError()
+
+  const { baseUrl, apiKey, timeout, cacheSeconds, ...params } = config
+
+  const url = baseUrl || "https://api.currencybeacon.com/v1"
+  const cacheKey = `${CacheKeys.CurrentTicker}:currencybeacon:${base}:*`
+
+  const getCachedRates = async (): Promise<CurrencyBeaconRates | null> => {
```
Comment (vindard) [inline realtime/src/services/exchanges/currencybeacon/index.ts:28]: Why `null` and not `undefined`?

**Review by vindard — COMMENTED** [review]
Small comments, but otherwise looks ok to me

File: realtime/src/services/exchanges/currencybeacon/index.ts (hunk 1 above)
Comment (dolcalmi) [reply realtime/src/services/exchanges/currencybeacon/index.ts]: fixed https://github.com/GaloyMoney/price/pull/36/commits/cfdd36a3a9a658db89bad689231b833480a3a69d

File: realtime/src/services/exchanges/currencybeacon/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        response.status > 400 ||
+        !response.data.response ||
+        !response.data.response.rates ||
+        !Object.keys(response.data.response.rates).length
+      )
+        return new UnknownExchangeServiceError(
+          `Invalid response. Error ${response.status}`,
+        )
+
+      const rates = response.data.response.rates
```
[the remark this answers is not in the document]
Comment (dolcalmi) [reply realtime/src/services/exchanges/currencybeacon/index.ts]: we cant do a validation key by key (is dynamic ) but I added a validation (content and type), please check https://github.com/GaloyMoney/price/pull/36/commits/b81528a6b11d38f7d61f931aacff4b94f8782f2c

**Review by vindard — APPROVED** [review]

**Pull request merged**